### ISSUE TITLE
[Feature Request] disable CSP meta tag for SSG

### ISSUE THREAD

**Issue by mtdvlpr** [Baroshem/nuxt-security#339]
Since I'm now extracting the headers, including CSP, and importing them in my nginx config, I don't need the CSP meta tag that's generated. Is there a way to disable the meta tag but keep the CSP configuration?

_Originally posted by @mtdvlpr in https://github.com/Baroshem/nuxt-security/issues/336#issuecomment-1888034583_

**Comment by Baroshem** [comment]
Hey @mtdvlpr 

Thanks for rising this issue. As I mentioned in the comment I am not sure if there is a way to do that right now but we for sure could add a new configuration option:

```ts
security: {
  ssg: {
    enabled: boolean;
    hashScripts: boolean;
    // more
  }
}
```

Actually this requires to add a new entry to the module type, a default value, and an if statement in the place where we are registering these SSG plugins (this is just a rough estimation though).

Would you be open for contributing? Me or @vejja will provide any help needed to let you be part of Nuxt Security :)

**Comment by mtdvlpr** [comment]
@Baroshem sure! I'll give it a go when I'm home and create a PR with you as reviewers when ready.

**Comment by Baroshem** [comment]
Awesome, let me know if you would need any assistance. We are here to help :)

**Comment by mtdvlpr** [comment]
@Baroshem I could use a review :)

**Comment by vejja** [comment]
Quick comment: we won't generate headers anymore if we remove 05-cspSsgPresets.ts to solve issue #335 
Let's think about how we can deal with this

**Comment by vejja** [comment]
> @vejja I think that we just have to find a way to replace nuxt/kit with other imports is it correct?
> 
> I am not sure if we actually need to remove the cspSsgPresets plugin. Is it necessary?

I think it is necessary to remove it, or to replace it with something else
From what I understand from @pi0, the bug comes from the fact that we are invoking `useNitro()` from within a Nitro plugin runtime.

**Comment by Baroshem** [comment]
Damn, I really liked that feature :(

Would you be able to maybe try to refactor this plugin somehow to not use nuxt/kit? I will try to do something about it as well. Maybe there is a way to achieve this by not removing the code but to instead fix this behavior.

**Comment by mtdvlpr** [comment]
I really hope the cspSsgPreset plugin can stay!

**Comment by Baroshem** [comment]
PR was merged in 1.2.0 :)

Thanks for the amazing work!
